addons/binding/org.openhab.binding.camera/src/main/java/org/openhab/binding/camera/handler/CameraHandler.java

```diff
@@ -0,0 +1,195 @@
+ */
+package org.openhab.binding.camera.handler;
+
+import static org.openhab.binding.camera.CameraBindingConstants.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.AuthCache;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CameraHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Thomas Hartwig - Initial contribution
+ */
+public class CameraHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(CameraHandler.class);
+    private final AtomicBoolean refreshInProgress = new AtomicBoolean(false);
+    private final AtomicBoolean initialized = new AtomicBoolean(false);
+    private final ExecutorService serviceCached = Executors.newCachedThreadPool();
+    private String urlSnapshot;
+    private String urlUsername = "";
+    private String urlPassword = "";
+
+    public CameraHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_IMAGE)) {
+            if (command.toString().equals("REFRESH")) {
+                refreshData();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Initialize thing: {}::{}", getThing().getLabel(), getThing().getUID());
+        }
+        Object paramUrl = getConfig().get("urlSnapshot");
```

Comment:
This is already a `String` and can just be casted.

Reply:
OK